Core/include/Acts/Geometry/NavigationPolicyFactory.hpp

```diff
@@ -63,153 +35,128 @@ concept NavigationPolicyIsolatedFactoryConcept = requires(
-
-  /// Create a factory with the specified policy added
-  /// @tparam P The policy type to add
-  /// @param args The arguments to pass to the policy constructor
-  /// @note Arguments need to be copy constructible because the factory must be
-  ///       able to execute multiple times.
-  /// @return A new policy factory including the @c P policy.
-  template <NavigationPolicyConcept P, typename... Args>
-    requires(std::is_constructible_v<P, const GeometryContext&,
-                                     const TrackingVolume&, const Logger&,
-                                     Args...> &&
-             (std::is_copy_constructible_v<Args> && ...))
-  constexpr auto add(Args&&... args) && {
-    auto factory = [=](const GeometryContext& gctx,
-                       const TrackingVolume& volume, const Logger& logger) {
-      return P{gctx, volume, logger, args...};
-    };
-
-    return NavigationPolicyFactoryImpl<decltype(factory)>{
-        std::make_tuple(std::move(factory))};
-  }
+/// Base class for navigation policy factories. The factory can be assembled
+/// iteratively by using `make` followed by a number of calls to the `add`
+/// function of the helper type. Example:
+///
+/// ```cpp
+/// auto factory = NavigationPolicyFactory{}
+///  .add<NavigationPolicy1>(arg1, arg2)
+///  .add<NavigationPolicy2>(/*no args*/)
+///  .asUniquePtr();
+/// ```
+class NavigationPolicyFactory {
+ private:
+  /// Type alias for factory functions that create navigation policies
+  using factory_type = std::function<std::unique_ptr<INavigationPolicy>(
+      const GeometryContext&, const TrackingVolume&, const Logger&)>;
 
-  /// Create a factory with a policy returned by a factory function
-  /// @tparam Fn The type of the function to construct the policy
-  /// @param args The arguments to pass to the policy factory
-  /// @note Arguments need to be copy constructible because the factory must be
-  ///       able to execute multiple times.
-  /// @return A new policy factory including the function
-  template <typename Fn, typename... Args>
-    requires(NavigationPolicyIsolatedFactoryConcept<Fn, Args...>)
-  constexpr auto add(Fn&& fn, Args&&... args) {
-    auto factory = [=](const GeometryContext& gctx,
-                       const TrackingVolume& volume, const Logger& logger) {
-      return fn(gctx, volume, logger, args...);
-    };
+  /// Private constructor for internal use
+  /// @param factories Vector of factory functions
+  explicit NavigationPolicyFactory(std::vector<factory_type>&& factories)
+      : m_factories(std::move(factories)) {}
 
-    return NavigationPolicyFactoryImpl<decltype(factory)>{
-        std::make_tuple(std::move(factory))};
+ public:
+  /// Default constructor
+  NavigationPolicyFactory() = default;
+  /// Move constructor
+  NavigationPolicyFactory(NavigationPolicyFactory&& other) = default;
+  /// Copy constructor
+  NavigationPolicyFactory(const NavigationPolicyFactory& other) = default;
```

Comment:
I think this can be dropped? or copy and move assignment should also be defaulted. I think they are dropped right now?

Reply:
I can remove the defaulted copy and move constructors, but I need to keep the default constructor.